### PR TITLE
feat: button play surah audio at home page

### PR DESCRIPTION
**Pull request by jokot**
### Background

Add button play audio for each surah item at home page

### Impacted Products

- Home page
- Surah page

### How to Test

Check on impacted page

### Video

https://github.com/Yaumi-Digital-School/quranplus-flutter/assets/25982672/52570140-3b81-48d5-af98-6ffa9fc936d3

### Pre-Deployment Checklist

- [ ] Run Local OK (Mandatory)
- [ ] Run Dev Build on Real Device

### Test Checklist

- [ ] Unit Testing
- [ ] Other (Please Elaborate)

### Code Review Guidelines

https://oyteam.quip.com/4m8kAafMWWiX/Code-Review-Manifesto

### CODE REVIEW

File: lib/pages/home_page_v2/home_page_state_notifier.dart [hunk 1]
```diff
@@ -114,6 +131,32 @@ class HomePageStateNotifier extends BaseStateNotifier<HomePageState> {
...
+      ),
+      _audioApi,
+      _audioPlayer,
+    );
+
+    state = state.copyWith(
+      audioSuratLoaded: null,
+    );
+
+    callback();
```
Comment (indraseptama) [inline lib/pages/home_page_v2/home_page_state_notifier.dart:157]: eh sorry ini fungsi buat apa ya?

File: lib/pages/home_page_v2/home_page_v2.dart [hunk 2]
```diff
@@ -519,119 +524,227 @@ class ListSuratByJuz extends StatelessWidget {
...
+                                  onPressed: () async {
+                                    notifier.playOnAyah(
+                                      surats[index],
+                                      () async {
+                                        int page = surats[index].startPageToInt;
+                                        int startPageInIndexValue = page - 1;
+
+                                        final dynamic param =
+                                            await Navigator.pushNamed(
+                                          context,
```
Comment (indraseptama) [inline lib/pages/home_page_v2/home_page_v2.dart:694]: Wrap the logic to navigate to another page within a method kot

File: lib/pages/home_page_v2/home_page_state_notifier.dart [hunk 3]
```diff
@@ -114,6 +131,51 @@ class HomePageStateNotifier extends BaseStateNotifier<HomePageState> {
     }
   }
 
+  Future<void> initAyahAudio(
+    SuratByJuz surat,
+    Function() onSuccess,
+    Function() onLoadError,
+    Function() onPlayBackError,
```
Comment (yumnapratista24) [inline lib/pages/home_page_v2/home_page_state_notifier.dart:138]: @jokot kot, bisa bikin ini jadi named parameter kah ? soalnnya udah ada 4 parameter, agak sulit untuk bedain takutnya nanti

**Review by yumnapratista24 — APPROVED** [review]
lgtm

**Pull request merged**